editoast/src/schema/rolling_stock.rs

```diff
@@ -56,20 +60,44 @@ pub struct RollingStock {
     power_restrictions: Option<JsonValue>,
 }
 
+#[derive(Debug, Serialize)]
+pub struct RollingStockWithLiveries {
+    #[serde(flatten)]
+    pub rolling_stock: RollingStock,
+    pub liveries: Vec<RollingStockLiveryMetadata>,
+}
+
 impl RollingStock {
-    pub async fn retrieve(db_pool: Data<DbPool>, rolling_stock_id: i64) -> Result<RollingStock> {
-        block::<_, Result<RollingStock>>(move || {
+    pub async fn retrieve(
+        db_pool: Data<DbPool>,
+        rolling_stock_id: i64,
+    ) -> Result<RollingStockWithLiveries> {
+        block::<_, Result<RollingStockWithLiveries>>(move || {
+            use crate::tables::osrd_infra_rollingstock::dsl as rolling_stock_dsl;
+            use crate::tables::osrd_infra_rollingstocklivery::dsl as livery_dsl;
             let mut conn = db_pool.get().expect("Failed to get DB connection");
-            match dsl::osrd_infra_rollingstock
+            let rolling_stock = match rolling_stock_dsl::osrd_infra_rollingstock
                 .find(rolling_stock_id)
                 .first(&mut conn)
             {
-                Ok(rolling_stock) => Ok(rolling_stock),
+                Ok(rolling_stock) => Ok::<_, InternalError>(rolling_stock),
                 Err(DieselError::NotFound) => {
                     Err(RollingStockError::NotFound { rolling_stock_id }.into())
                 }
                 Err(e) => Err(e.into()),
-            }
+            }?;
+            let liveries = match livery_dsl::osrd_infra_rollingstocklivery
+                .filter(livery_dsl::rolling_stock_id.eq(rolling_stock_id))
+                .select(RollingStockLiveryMetadata::as_select())
+                .load(&mut conn)
+            {
+                Ok(liveries) => Ok(liveries),
+                Err(e) => Err(InternalError::from(e)),
+            }?;
+            Ok(RollingStockWithLiveries {
+                rolling_stock,
+                liveries,
+            })
```

Comment:
```suggestion
            match livery_dsl::osrd_infra_rollingstocklivery
                .filter(livery_dsl::rolling_stock_id.eq(rolling_stock_id))
                .select(RollingStockLiveryMetadata::as_select())
                .load(&mut conn)
            {
                Ok(liveries) => Ok(RollingStockWithLiveries {
                    rolling_stock,
                    liveries,
                }),
                Err(e) => Err(InternalError::from(e)),
            }
            })
```

Reply:
This can be simplified again using simply `?`